### PR TITLE
Add basic functionality to add, delete, and complete todo items

### PR DESCRIPTION
**Pull request by leehornsoftwire**
module_1 pull request

### CODE REVIEW

File: todo_app/data/session_items.py [hunk 1]
```diff
@@ -1,8 +1,75 @@
...
+
+    def save(self):
+        session["_items_by_id"] = self._items_by_id
+        session["_next_item_id"] = self._next_item_id
+
+    def load() -> Optional[Items]:
+        next_item_id = int(session.get("_next_item_id"))
+        if next_item_id is None:
+            return None
+        items_by_id_as_strings: Dict[str, Dict[str, str]] = session.get("_items_by_id")
```
Comment (JoshAdamPowell) [inline todo_app/data/session_items.py:63]: This code doesn't handle the case where we have no tasks - it causes the page to fail. I think the solution is to provide reasonable defaults, or maybe return early.

File: todo_app/templates/index.html [hunk 2]
```diff
@@ -11,8 +11,30 @@ <h1 class="display-4">To-Do App</h1>
...
+            <input type="hidden" name="id" value={{ id }}>
+            <input type="submit" value="Delete">
+          </form>
+        {% endfor %}
       </ul>
     </div>
   </div>
+  <form action="/items/new" method="post">
+    <label for="title">Title:</label>
+    <input type="text" id="title" name="title">
```
Comment (JoshAdamPowell) [inline todo_app/templates/index.html:37]: minor but might be nice for this to have a `required` attribute

**Review by JoshAdamPowell — CHANGES_REQUESTED** [review]
Looks good apart from the fact that it fails for the empty case 😅. Please fix that before we merge this!

**Comment by JackMead** [comment]
Thanks for fixing that up, happy it's now merged 😄

**Pull request merged**